tests/test_cli_compile.py

```diff
@@ -2190,6 +2190,49 @@ def test_multiple_extras(fake_dists, runner, make_module, fname, content, extra_
     assert "extra ==" not in out.stderr
 
 
+@pytest.mark.network
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_build_system_requires(
+    request, fake_dists, runner, make_module, fname, content
+):
+    """
+    Test that requirements needed to build a package are included
+    """
+    meta_path = make_module(fname=fname, content=content)
+    out = runner.invoke(
+        cli,
+        [
+            "-n",
+            "--find-links",
+            fake_dists,
+            "--allow-unsafe",
+            "--no-annotate",
+            "--no-emit-options",
+            "--no-header",
+            "--build-system-requires",
+            meta_path,
+        ],
+    )
+    assert out.exit_code == 0, out.stderr
+    assert "small-fake-a==0.1" in out.stderr
+    assert "small-fake-b==0.2" in out.stderr
+    test_case_id = request.node.callspec.id.split("-")[1]
+    if test_case_id == "flit":
+        assert "flit-core==" in out.stderr
+        assert "poetry-core==" not in out.stderr
+        assert "setuptools==" not in out.stderr
+    elif test_case_id == "poetry":
+        assert "flit-core==" not in out.stderr
+        assert "poetry-core==" in out.stderr
+        assert "setuptools==" not in out.stderr
+    elif test_case_id in {"setup.py", "setup.cfg"}:
+        assert "flit-core==" not in out.stderr
+        assert "poetry-core==" not in out.stderr
+        assert "setuptools==" in out.stderr
+    else:
+        pytest.fail(f"Test not adapted for {test_case_id}")
```

Comment:
These should be different test cases in parametrize. Declare the expectations there instead of making the test dynamic and branchy.